### PR TITLE
feat(cdsctl): generate exportentities json schema

### PR DESCRIPTION
**Pull request by richardlt**
1. Description
- [x] generate json schema.
- [x] generate config for vscode.
- [x] generate action builtin doc with example.
- [x] add more description for json schema fields.
- [x] add required field for json schema.
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#9303.0 ✘  
* Build minimal
  * cdsctl linux ✘  
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#9312.0 ✘  
* Build minimal
  * cdsctl linux ✘  
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#9369.0 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✘  
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report ut-engine#9369.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#9370.0 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✘  
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report ut-engine#9370.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#9371.0 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✘  
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report ut-engine#9371.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report it#9377.0 ✘  
* w-it
  * w-it ✘

File: docs/actions/builtin-serve-static-files.part.md [hunk 1]
```diff
@@ -29,3 +11,7 @@ Pay attention this action is only available if your objectstore is configured to
 * View your static files with links in the tab artifact
 
 ![img](/images/workflows.pipelines.actions.builtin.serve-static-files-tab.png)
+
+## Notes
+
+Pay attention this action is only available if your objectstore is configured to use Openstack Swift. And fow now by default your static files will be deleted after 2 months.
```
Comment (bnjjj) [inline docs/actions/builtin-serve-static-files.part.md]: Maybe also available for S3 now ? @yesnault ?

File: sdk/exportentities/doc_test.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+	"github.com/ovh/cds/sdk/exportentities"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
+
+	"github.com/ovh/cds/sdk"
+)
+
+func TestPipeline(t *testing.T) {
+	/*	exportentities.PipelineV1{
```
Comment (bnjjj) [inline sdk/exportentities/doc_test.go]: to delete ?

File: docs/actions/builtin-serve-static-files.part.md (hunk 1 above)
Comment (yesnault) [reply docs/actions/builtin-serve-static-files.part.md]: yes

File: sdk/exportentities/step.go [hunk 3]
```diff
@@ -0,0 +1,731 @@
...
+	return &a, nil
+}
+
+func (s Step) isScript() bool { return s.Script != nil }
+
+func (s Step) asScript() (sdk.Action, error) {
+	var a sdk.Action
+	// TODO use typed value for script
+	/*
+		val := strings.Join(*s.Script, "\n")
```
Comment (bnjjj) [inline sdk/exportentities/step.go]: ?

File: sdk/exportentities/step.go [hunk 4]
```diff
@@ -0,0 +1,731 @@
...
+	*/
+
+	var val string
+	if script, ok := s.Script.([]interface{}); ok {
+		lines := make([]string, len(script))
+		for i := range script {
+			if line, ok := script[i].(string); ok {
```
Comment (bnjjj) [inline sdk/exportentities/step.go]:
```suggestion
			if line, okString := script[i].(string); okString {
```

File: sdk/exportentities/step.go [hunk 5]
```diff
@@ -0,0 +1,731 @@
...
+	return a, nil
+}
+
+func (s Step) isArtifactUpload() bool { return s.ArtifactUpload != nil }
+
+func (s Step) asArtifactUpload() (sdk.Action, error) {
+	var a sdk.Action
+	// TODO use type for artifact upload
+	/*
+			m, err := stepToMap(s.ArtifactUpload)
```
Comment (bnjjj) [inline sdk/exportentities/step.go]: ?

**Comment by ovh-cds** [comment]
CDS Report ut-engine#9521.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report build-ui#9527.0 ✘  
* Compile
  * Build UI ✔ 
  * TU All -   
  * TU model ✔ 
  * TU service ✔ 
  * TU shared ✔ 
  * TU store ✔ 
  * TU views ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#9527.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Pull request merged**